### PR TITLE
Setting up Solr server using authentication and updating the python code to access solr using user and password

### PR DESCRIPTION
**Pull request by liseli**
This PR updates the docker-compose and the Python code to start up and access to a Solr server using authentication. 

The pull request includes:

Updating documentation
Fixing all the tests.
We are updating the Python code to access Solr with authentication.
Updating docker file and docker-compose to start up Solr with authentication 
In the docker-compose has been included a service to index some data in Solr (Useful for testing)

**How do we test this PR?**

```
git checkout readme_improvements
./init.sh 
docker compose up -d
```

**Check if the API is running Access to the URL:** http://localhost:8000/docs

**Run the tests**
`docker compose exec full_text_searcher python -m pytest`

You will see the following output if all the tests ran without issue

image

**[Optional] In case you want to run any of the implemented features**

**Run the use case 1: Query the Solr index given a phrase**

```
docker compose exec full_text_searcher python ht_full_text_search/ht_full_text_searcher.py --env dev --query_string "persons and corporations" --query_config ocronly
```

Output: In your terminal, you should see the list of documents obtained with the input query.

image

**Run the Use case 2: Query Solr given a list of phases (included in list_query_file.txt) and generate an CSV file with the query results.**

```
docker compose exec full_text_searcher python scripts/generate_query_results_in_batch.py \  
              --env dev \
              --list_phrase_file scripts/list_query_file.txt \
              --query_config ocronly
```

Inside the folder scripts/query_results some CSV files should be generated for each of the queries included in list_query_file.txt

**Run the Use Case 4: Query solr using the API**

query endpoint: `curl --location 'http://localhost:8000/query/?query=persons%20and%20corporations&env=dev' --form 'query="'\''\"persons and corporations\"'\''"'`

Output:

`{"id": "chi.67160761", "author": ["Hawaii. Public Utilities Commission"], "title": ["Annual report"]}{"id": "coo.31924007104676", "author": ["Wisconsin. Railroad Commissioners' Department."], "title": ["Biennial report of the Railroad Commissioner of the State of Wisconsin"]}`

status endpoint: `curl --location 'http://localhost:8000/ping/'`
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,21 +60,27 @@ The API is based on the [FastAPI](https://fastapi.tiangolo.com/) library.
...
 * Code Editor
-* Python 3 and Poetry (If you want to run the application in your local environment)
+* Python 3 and Poetry (If you want to run the application in your local environment). See the installation section below.
   * To access to prod Solr server, you need it, 
     * to have a VPN connection to the HathiTrust network
     * to set up an ssh tunnel `ssh -L8081:localhost:8081 squishee-1.umdl.umich.edu`
     * to run the application in your local environment with the parameter `--env prod`.We don't have an 
     acceptable alternative, nor is it necessary to set up access to the production server via a Docker file.
-  * To query production, you will have to run the application locally and open and ssh connection to squishee-1. 
+  * To query production, you will have to run the application locally and open and ssh connection to squishee-1.
```
Comment (aelkiss) [inline README.md:82]: squishee-1 will be retired, so we should list the new one (`macc-ht-solr-lss-1`) or reference some external documentation in confluence

File: docker-compose.yml [hunk 2]
```diff
@@ -42,35 +42,72 @@ services:
...
+    image: ghcr.io/hathitrust/full-text-search-cloud:shards-docker
     container_name: solr-lss-dev
     ports:
-        - "8983:8983"
+      - "8983:8983"
+    environment:
+      ZK_HOST: zoo1:2181
+      SOLR_OPTS: -XX:-UseLargePages
+    env_file:
+      - .env
```
Comment (aelkiss) [inline docker-compose.yml:64]: docker-compose references this `.env` as well as the solr user& password env vars, but the `.env` file isn't included with sample values, and I didn't see anything in the instructions for setting that up.

File: solr_dataset/chi096189208_solr_full_text.xml
Comment (aelkiss) [inline solr_dataset/chi096189208_solr_full_text.xml:1]: Just to check, have we verified that all these sample files are public domain?

**Review by aelkiss — COMMENTED** [review]
Changes to support authentication look good to me. I am not sure if there are some instructions or files that may be missing so that `SOLR_USER` and `SOLR_PASSWORD` are set in the docker compose environment -- see my comments.

File: docker-compose.yml (hunk 2 above)
Comment (liseli) [reply docker-compose.yml:64]: .env file is generated from the .env.example that is included in the repository when the script ./init.sh is executed. I will add this information to the documentation. I have a question: Using this logic to store the variables in a .env file will require some change on our GitHub actions? This repository does not have GitHub actions yet. I will add a task for that.

Comment (aelkiss) [reply docker-compose.yml:64]: I don't think it should require any change in the github actions so long as the test action runs init.sh

Comment (liseli) [reply docker-compose.yml:64]: Ok, I should figure out how to use the init.sh file in the GitHub action. Probably we already do that on some repository I could use as a reference.

**Review by moseshll — APPROVED** [review]
The addition of `service_completed_successfully` to the data loader dependency allows tests to pass for me. I would recommend adding a newline to the end of `env.example`, just to suppress warnings.

**Pull request merged**